projects/week_1/Instructions.md

```diff
@@ -0,0 +1,350 @@
+
+Starting on the line after your `new_tags` declaration, add a for loop that iterates over `tag_array_msg->tags`.
+
+**Tip:** You can get the count of elements in a `std::vector` by calling its `size()` method.
+
+<details>
+<summary><b>Hint:</b> For loops</summary>
+<p>You can write a for loop that iterates from `i=0` to `i=9` like this:</p>
+<pre><code>for(int i = 0; i < 10; ++i)
+{
+   // body of loop
+}</code></pre>
+</details>
+
+In the body of this new loop, declare a variable named `new_tag` of type `stsl_interfaces::msg::Tag`. Then, copy the `id` member of the current old tag into `new_tag.id` like this:
+
+```c++
+new_tag.id = tag_array_msg->tags[i].id;
+```
+
+**Tip:** `i` in the above snippet should be replaced with whatever you named the variable being incremented in your for loop.
+
+Finally, at the end of the loop body use `push_back()` to add `new_tag` to the `new_tags` vector.
+
+### 2.7 Transform tag position
+
+We now have two transformations we need to apply to our tags. The first we grabbed for you from the ROS TF system (which we'll cover in detail later). This is `camera_to_base_transform`, which transforms things from the camera's conventional frame to the robot's base frame. The second transform, `camera_optical_to_conventional_transform` is the rotation transformation you created in the `getTransformationMatrixForOpticalFrame()` function. Our objective is to transform all of the tag detections from the camera's optical frame to the robot's base frame, so we need to apply both of these transformations to the position and orientation of each tag. In this section, we'll handle the position.
+
+Let's start by creating a homogeneous vector for our tag's position. In the body of our for loop, after the `new_tag.id = tag_array_msg->tags[i].id;` line, declare a variable of type `Eigen::Vector4d`, named `position`. Initialize `position` by passing, to its constructor, the x, y, and z coordinates from the old tag's pose, and 1 for the final element. That will look like this:
+
+```c++
+Eigen::Vector4d position(
+  tag_array_msg->tags[i].pose.position.x,
+  tag_array_msg->tags[i].pose.position.y,
+  tag_array_msg->tags[i].pose.position.z,
+  1
+);
+```
+
+Now, apply the transformations by multiplying the transformation objects, `camera_to_base_transform` and `camera_optical_to_conventional_transform`, with `position`. Assign the result of that multiplication back to `position`. Remember, the order of this multiplication matters. Be sure to apply `camera_optical_to_conventional_transform` before `camera_to_base_transform`.
+
+<details>
+<summary><b>Hint:</b> Ordering transformations</summary>
+<p>Transforms that are applied first should appear on the right of a multiplication chain. To apply A, then B, then C, we would write <code>C * B * A</code>.</p>
+</details>
+
+The last thing we need to do for this section is to copy the new position's coordinates into the `new_tag` message. Copying the x value looks like this:
```

Comment:
Should this be up where we first discuss getTransformationMatrixForOpticalFrame()?

Reply:
It can be if we want. Right now, the optical frame instructions just tell them the order to use.

Reply:
It's fine either way